src/Library.php

```diff
@@ -79,7 +74,57 @@ public static function maybe_create_tables() {
 			"
 		);
 
-		update_option( 'wc_blocks_db_version', \Automattic\WooCommerce\Blocks\Package::get_version() );
+		if ( ! $exists ) {
+			return self::add_create_table_notice( $table_name );
+		}
+
+		// Update succeeded. This is only updated when successful and validated.
+		// $schema_version should be incremented when changes to schema are made within this method.
+		update_option( 'wc_blocks_db_schema_version', $schema_version );
+	}
+
+	/**
+	 * Create database table, if it doesn't already exist.
+	 *
+	 * Based on admin/install-helper.php maybe_create_table function.
+	 *
+	 * @param string $table_name Database table name.
+	 * @param string $create_sql Create database table SQL.
+	 * @return bool False on error, true if already exists or success.
+	 */
+	protected static function maybe_create_table( $table_name, $create_sql ) {
+		global $wpdb;
+
+		if ( in_array( $table_name, $wpdb->get_col( 'SHOW TABLES', 0 ), true ) ) {
+			return true;
+		}
+
+		$wpdb->hide_errors();
+		$wpdb->query( $create_sql ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+
+		return in_array( $table_name, $wpdb->get_col( 'SHOW TABLES', 0 ), true );
+	}
```

Comment:
I'm curious, why not just use `maybe_create_table` directly? If it's because `maybe_create_table` is a pluggable function I don't think we should worry about that, there may be legit cases where it's been overridden (and we're not really responsible for those cases imo).  If it's because we're hiding errors here, we can still call `$wpdb->hide_errors` before calling `maybe_create_table` (and return to previous value of `$wpdb->show_errors` after calling).

Reply:
Because I did some searching and found examples of some people removing the file where this function is found...after installing WordPress. The require seemed risky. 

Since it's pluggable also, and fairly small, I thought it best to include it to minimise chance of conflict.

Reply:
It looks like `maybe_create_table` is also defined in `wp-admin/includes/upgrade.php` which is probably a more reliable file to require?

Reply:
nah scratch that, it means loading a bunch of other files on every request too (which could be problematic). Let's just roll with what you have (but still address the hidden errors comment I made).